wizeline/remix-project-lab
# Feat/218/allow non Wizeline users to request join a project

# What does this PR do?
This "PR" allows users to apply to projects on the "IntershipProjects" page, once they apply the information is displayed in "applicants".



# Where should the reviewer start?
app/routes/internshipProjects/$projectId/index.tsx
app/routes/internshipProjects/$projectId/appliedproject.tsx
app/models/applicant.server.ts
app/routes/applicants/$applicantId/index.tsx
app/routes/projects/$projectId/index.tsx



# How should this be manually tested?
1. Log in through Linkdin, fill in the applicant form and you will be redirected to the "intershipProjects" page.
2.Select any of the displayed projects
3.Click on the "apply" button and you will be redirected to the "intershipProjects" page. (You can repeat this process with the other projects).
4.Verify that if you enter a project to which you have already applied the "apply" button is disabled.
5.Login to your wizeline account, and go to "/applicants".
6.Select the applicant and verify that in the "Applied projects" section appear the projects to which he/she applied. You can also add the "Applied projects" column in "/applicants" to see the projects to which you have applied. 
7.With your Wizeline account, verify that when you enter a project, in the "applicants" section, the applicants that apply to that project appear.



# Any background context you want to provide?



# What are the relevant tickets?
#218  



# Screenshots
Before applying: 
<img width="1728" alt="Captura de pantalla 2023-09-29 a la(s) 2 59 56 p m" src="https://github.com/wizeline/remix-project-lab/assets/58340818/357cb81a-52a4-40eb-a462-a63e6edb1636">

After applying:
<img width="1728" alt="Captura de pantalla 2023-09-29 a la(s) 2 59 49 p m" src="https://github.com/wizeline/remix-project-lab/assets/58340818/75f27632-9678-4288-a920-4c6ccb5b2c0f">

Applicant column:
<img width="1728" alt="Captura de pantalla 2023-09-29 a la(s) 2 58 13 p m" src="https://github.com/wizeline/remix-project-lab/assets/58340818/5353b757-0ff4-429e-8b13-56701a26d04e">

Show detailed information of applied projects:
<img width="1269" alt="Captura de pantalla 2023-10-04 a la(s) 12 38 37 p m" src="https://github.com/wizeline/remix-project-lab/assets/58340818/3b79d7e0-d298-4a89-a006-7b8d285d0137">

Show applicants in projectid page:
<img width="1269" alt="Captura de pantalla 2023-10-04 a la(s) 12 38 50 p m" src="https://github.com/wizeline/remix-project-lab/assets/58340818/878906aa-401a-4d3a-b211-581e6e9533a4">




# Questions



# Checklist
